pkg/quota/image/imagestream_evaluator.go

```diff
@@ -31,13 +33,21 @@ func NewImageStreamEvaluator(isNamespacer osclient.ImageStreamsNamespacer) kquot
 		ConstraintsFunc:      generic.ObjectCountConstraintsFunc(imageapi.ResourceImageStreams),
 		UsageFunc:            generic.ObjectCountUsageFunc(imageapi.ResourceImageStreams),
 		ListFuncByNamespace: func(namespace string, options kapi.ListOptions) ([]runtime.Object, error) {
-			itemList, err := isNamespacer.ImageStreams(namespace).List(options)
+			list, err := store.ImageStreams(namespace).List(options.LabelSelector)
 			if err != nil {
 				return nil, err
 			}
-			results := make([]runtime.Object, 0, len(itemList.Items))
-			for i := range itemList.Items {
-				results = append(results, &itemList.Items[i])
+			results := make([]runtime.Object, 0, len(list))
+			for _, is := range list {
+				objCopy, err := kapi.Scheme.DeepCopy(is)
```

Comment:
You should deep-copy when you are about to mutate but I am not sure if we gain much. I guess it depends on how much we mutate below this point.